vendor/plist_to_html/plist_to_html/PlistToHtml.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+import argparse
+import json
+import os
+import plistlib
+import shutil
+from xml.parsers.expat import ExpatError
+
+
+def get_file_content(filename):
+    with open(filename, 'r') as f:
+        return f.read()
+
+
+class HtmlBuilder:
+    """
+    Helper class to create html file from a report data.
+    """
+    def __init__(self, layout_dir):
+        self.layout_dir = layout_dir
+
+        # Mapping layout tags to files.
+        self._layout_tag_files = {
+            'STYLE_CSS': 'style.css',
+            'CODEMIRROR_CSS': 'codemirror.min.css',
+            'CODEMIRROR_JS': 'codemirror.min.js',
+            'CLIKE_JS': 'clike.min.js',
+            'BUG_VIEWER': 'bugviewer.js'
+        }
+
+        # Get the HTML layout file content.
+        self._layout = get_file_content(
+            os.path.join(self.layout_dir, 'layout.html'))
+
+        # Get the content of the HTML layout dependecies.
+        self._tag_contents = {}
+        for tag in self._layout_tag_files:
+            self._tag_contents[tag] = get_file_content(
+                os.path.join(self.layout_dir, self._layout_tag_files[tag]))
+
+    def create(self, output_path, report_data):
+        """
+        Create html file with the given report data to the output path.
+        """
+        content = self._layout.replace('<$REPORT_DATA$>',
+                                       json.dumps(report_data))
+
+        for tag in self._layout_tag_files:
```

Comment:
Okay, I was thinking about this this morning. Perhaps, just for optimisation purposes if someone has a large payload of `plist`s to export, I would add an extra caching logic into this class.

 1. Load the layout file.
 2. Load the resources, and automatically replace these in the string that contains the HTML skeleton. And store this, in which only "dynamic" value positions are kept.
 3. Use this skeleton to, in the `create` method, only replace the dynamics into actual content and write them out.